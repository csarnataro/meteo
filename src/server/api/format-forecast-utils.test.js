const { DateTime } = require('luxon')

const {
  limitTo,
  filterHoursModulus3
} = require('./format-forecast-utils')

describe('limitTo', () => {
  it('should limit to 3 items', () => {
    const output = limitTo(3)([1, 2, 3, 4])
    expect(output).toEqual([1, 2, 3])
  })
  it('should not limit when size == items.length', () => {
    const output = limitTo(4)([1, 2, 3, 4])
    expect(output).toEqual([1, 2, 3, 4])
  })
  it('should not limit when size > items.length', () => {
    const output = limitTo(45)([1, 2, 3, 4])
    expect(output).toEqual([1, 2, 3, 4])
  })
})

describe('filterHoursModulus3', () => {
  it('should show only items with hour % 3', () => {
    const items = require('./__mocks__/api-darksky-forecast').hourly.data
    const output = filterHoursModulus3(items)
    expect(output.map(item => {
      const dt = DateTime.fromSeconds(item.time)
      dt.setLocale('it')
      return dt.toFormat('HH:mm')
    })).toMatchSnapshot()
  })
})
