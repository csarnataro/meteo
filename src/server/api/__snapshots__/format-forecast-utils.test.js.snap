// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterHoursModulus3 should show only items with hour % 3 1`] = `
Array [
  "09:00",
  "12:00",
  "15:00",
  "18:00",
  "21:00",
  "00:00",
  "03:00",
  "06:00",
  "09:00",
  "12:00",
  "15:00",
  "18:00",
  "21:00",
  "00:00",
  "03:00",
  "06:00",
  "09:00",
]
`;
